src/components/composer/composer/composer_store.ts

```diff
@@ -644,7 +644,11 @@ export class ComposerStore extends SpreadsheetStore {
       const pivotId = this.getters.getPivotIdFromPosition(position);
       const pivotCell = this.getters.getPivotCellFromPosition(position);
       const cell = this.getters.getCell(position);
-      if (pivotId && pivotCell.type !== "EMPTY" && !cell?.isFormula) {
+      const cellIsPivotFormula =
+        cell?.isFormula &&
+        this.getters.getFirstPivotFunction(position.sheetId, cell.compiledFormula.tokens)
+          ?.functionName === "PIVOT";
+      if (pivotId && pivotCell.type !== "EMPTY" && (!cell?.isFormula || cellIsPivotFormula)) {
```

Comment:
```suggestion
      const mainCell = getters.getCellSPreadingOn(position) // Forgot the exact name fo the getter
      const isDynamicPivotCell = this.getters.getFirstPivotFunction(position.sheetId, cell.compiledFormula.tokens)
          ?.functionName === "PIVOT"
      if (pivotId && pivotCell.type !== "EMPTY" && isDynamicPivotCell ) {
```
probably easier to understand the goal of all of this ? In the end the only thing we care about if if the pivot formula is dynamic or not no ?